static/js/containers/Payment_test.js

```diff
@@ -123,14 +139,34 @@ describe('Payment container', () => {
       });
     });
 
-    it('does not show the name of the user if it is blank', () => {
+    it('does not show the welcome message if the name of the user is blank', () => {
       SETTINGS.user.full_name = '';
       klassesStub = fetchStub
         .withArgs(klassesUrl)
         .returns(Promise.resolve(generateFakeKlasses(1)));
       return renderFullPaymentPage().then((wrapper) => {
-        let welcomeMessage = wrapper.find(welcomeMessageSelector);
-        assert.equal(welcomeMessage.text(), "Welcome to MIT Bootcamps");
+        assert.isFalse(wrapper.find(welcomeMsgSelector).exists());
+      });
+    });
+
+    it('shows payment due date message', () => {
+      [
+        moment().format(),
+        null
+      ].forEach((deadlineDateISO) => {
+        let fakeKlasses = generateFakeKlasses(1);
+        fakeKlasses[0].deadline_date = deadlineDateISO;
+        klassesStub = fetchStub.withArgs(klassesUrl)
+          .returns(Promise.resolve(fakeKlasses));
+        store.dispatch(setSelectedKlassIndex(0));
+
+        return renderFullPaymentPage().then((wrapper) => {
+          let deadlineText = wrapper.find(deadlineMsgSelector).text();
```

Comment:
Ah, I think the issue is that there's the for loop above but it would only have one iteration before getting to the return statement here.